.github/workflows/unit_tests.yml

```
@@ -0,0 +1,64 @@
+          python -m pip install -r requirements-dev.txt
+      - name: Run unittests
+        run: |
+          pytest --cov=ovos_utils --cov-report xml test/
+          # NOTE: additional pytest invocations should also add the --cov-append flag
+          #       or they will overwrite previous invocations' coverage reports
+          #       (for an example, see OVOS Skill Manager's workflow)
+      - name: Upload coverage
+        if: "${{ matrix.python-version == '3.12' }}"
+        env:
+          CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
+        uses: codecov/codecov-action@v2
```

actionlint (1.7.4)

the runner of "codecov/codecov-action@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)